src/CryptoNoteProtocol/CryptoNoteProtocolHandler.cpp

```diff
@@ -933,9 +995,34 @@ void CryptoNoteProtocolHandler::relay_block(NOTIFY_NEW_BLOCK::request& arg) {
   }
 }
 
-void CryptoNoteProtocolHandler::relay_transactions(NOTIFY_NEW_TRANSACTIONS::request& arg) {
-  auto buf = LevinProtocol::encode(arg);
-  m_p2p->externalRelayNotifyToAll(NOTIFY_NEW_TRANSACTIONS::ID, buf, nullptr);
+void CryptoNoteProtocolHandler::relay_transactions(NOTIFY_NEW_TRANSACTIONS::request& arg) { 
+  if (arg.stem > 0 && !m_dandelion_stem.empty()) { // Dandelion broadcast
+    std::mt19937 rng = Random::generator();
+    std::uniform_int_distribution<> dis(0, 100);
+    auto coin_flip = dis(rng);
+    if (coin_flip < CryptoNote::parameters::DANDELION_STEM_TX_PROPAGATION_PROBABILITY) { // Stem propagation
+      for (const auto& dandelion_peer : m_dandelion_stem) {
+        arg.stem--;
+        if (dandelion_peer.m_state == CryptoNoteConnectionContext::state_normal || dandelion_peer.m_state == CryptoNoteConnectionContext::state_synchronizing) {
```

Comment:
We can rewrite here, if peer is dead, then regenerate stem peers and forward to new stem peers

Reply:
Yes, an additional check is needed so as not to break the logic. There is a chance of failure if the node is unavailable.

Reply:
Not a big deal, it will just fluff tx. We randomly with 10% of chances do this anyways.

Reply:
Yep, you're right.